### PR TITLE
Feature/add or remove friends

### PR DESCRIPTION
**Pull request by MTCawthray**
# Thank you for pushing your code!
## What (if any) features are you implementing?
- This branch adds the ```add user``` and ```delete user``` mutations and allows a user to 'friend' or 'unfriend' another user.
- When the user views another user's profile, they can click the add user to friends button to, you guessed it, add that user to their friends list! If they are viewing the profile of a friend, the button will be a remove user button.
- Now when they have clicked to add a friend, the user will see that new friend displayed when they click the friends button in the nav bar in header. When they click the unfriend button, they will no longer show up on the friend list.

## What (if anything) did you refactor?
- I refactored the addFriend mutation to include a refetchQueries hook.

## Were there any issues that arose?
- I battled for a long time with updating the GET_USER query once the mutation was made. 
- I had a very hard time referencing the friend list for the user because of async issues. I found the ```onCompleted``` hook that helped with this! 

## Is there anything that you need from your teammates?
- NA

## Any other comments, questions, or concerns?
-  This gets us pretty darn close to reaching our MVP as far as functionality goes. 

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor

## How Has This Been Tested?
- [x] open index.html
- [x] console.log()
- [x] dev tools
- [ ] Cypress

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/Profile/Profile.js [hunk 1]
```diff
@@ -1,88 +1,128 @@
...
+    refetchQueries: [{ 
+      query: GET_USER, 
+      variables: { 
+        id: userId
+      }
+    }]
+  });
+
+
+  const friendButton = () => {
```
Comment (nataliemcintyre2021) [inline src/components/Profile/Profile.js:45]: Nice getting this working with this conditional logic.

File: src/components/DisplayList/DisplayList.js [hunk 2]
```diff
@@ -1,13 +1,23 @@
 import React from 'react';
 import Cards from '../Cards/Cards';
+import { useQuery } from '@apollo/client';
+import { GET_USER } from '../../graphQL/queries';
 import './DisplayList.css';
 
-const DisplayList = ({ friends, filteredUsers}) => {
+const DisplayList = ({ filteredUsers, id}) => {
+  const {error: getUserError, loading: getUserLoading, data: getUserData} = useQuery(GET_USER, {
```
Comment (hayleyw7) [inline src/components/DisplayList/DisplayList.js:8]: Great job! The useQuery here makes sense.

File: src/components/Profile/Profile.js [hunk 3]
```diff
@@ -1,88 +1,128 @@
...
+  const friendUser = () => {
+    addFriend({
+      variables: {
+        userId: userId,
+        friendId: profileView
+      }
+    })
+  }
+
+  const unfriendUser = () => {
```
Comment (nataliemcintyre2021) [inline src/components/Profile/Profile.js:77]: Naming conventions look great!

File: src/components/Profile/Profile.js [hunk 4]
```diff
@@ -1,88 +1,128 @@
...
-import {GET_USER} from '../../graphQL/queries';
-
-const Profile = ({ profileView, logOut, id }) => {
-
-  // const {myersBrigg, enneagram, username} = profileView
-  const {error, loading, data} = useQuery(GET_USER, {
-    variables: {id: profileView}
+import { useQuery, useMutation } from '@apollo/client';
+import { GET_USER } from '../../graphQL/queries';
+import { ADD_FRIEND, DELETE_FRIEND } from '../../graphQL/mutations';
```
Comment (hayleyw7) [inline src/components/Profile/Profile.js:8]: It works! Good mutation.

**Review by nataliemcintyre2021 — APPROVED** [review]

File: src/components/Profile/Profile.js [hunk 5]
```diff
@@ -1,88 +1,128 @@
...
+  const [addFriend, { error: addFriendError, loading: addFriendLoading }] = useMutation(ADD_FRIEND, {
+    refetchQueries: [{ 
+      query: GET_USER, 
+      variables: { 
+        id: userId
+      }
+    }]
+  });
+
+  const [deleteFriend, { error: deleteFriendError, loading: deleteFriendLoading }] = useMutation(DELETE_FRIEND, {
```
Comment (hayleyw7) [inline src/components/Profile/Profile.js:35]: Interesting how many variables are needed here. Tough work. Great job again.

**Review by hayleyw7 — APPROVED** [review]
Great job! Almost MVP?

**Pull request merged**